File: src/1346-Chack-If-N-and-Double-Exists/check_if_n_and_double_exists.py
from typing import List

class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        seen = set()
        for num in arr:
            if num * 2 in seen or (num % 2 == 0 and num // 2 in seen):
                return True
            seen.add(num)
        return False

if __name__ == "__main__":
    arr: List[int] = [10,2,5,3]
    
    solution = Solution()
    print(solution.checkIfExist(arr))
    

File: src/1346-Chack-If-N-and-Double-Exists/check_if_n_and_double_exists.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

class Solution {
public:
    bool checkIfExist(std::vector<int>& arr) {
        std::unordered_set<int> seen;
        for (int num : arr) {
            if (seen.count(num * 2) || (num % 2 == 0 && seen.count(num / 2))) {
                return true;
            }
            seen.insert(num);
        }
        return false;
    }
};

int main() {
    std::vector<int> arr = {10,2,5,3};

    Solution solution;

    std::cout << std::boolalpha << solution.checkIfExist(arr) << std::endl; 

    return 0;
}